src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -229,12 +238,56 @@ public void validate(ClientState state)
         validateDefaultTimeToLive(attrs.asNewTableParams());
     }
 
+    private void maybeCopyIndexes(TableMetadata.Builder builder, TableMetadata sourceTableMeta, KeyspaceMetadata targetKeyspaceMeta)
+    {
+        if (createLikeOption != CreateLikeOption.INDEXES || sourceTableMeta.indexes.isEmpty())
+            return;
+
+        Set<String> customIndexes = Sets.newTreeSet();
+        Set<String> alreadyExistingIndexes = Sets.newTreeSet();
+        List<IndexMetadata> indexesToCopy = new ArrayList<>();
+        for (IndexMetadata indexMetadata : sourceTableMeta.indexes)
+        {
+            // only sai and legacy secondary index is supported
+            if (indexMetadata.kind == IndexMetadata.Kind.CUSTOM &&
+                indexMetadata.getIndexClassName() != StorageAttachedIndex.class.getCanonicalName())
```

Comment:
I do not think this is necessary. All we need to check is if that index is `CUSTOM`.